src/components/views/rooms/RoomTile.tsx

```diff
@@ -154,6 +155,22 @@ export default class RoomTile extends React.Component<IProps, IState> {
         MessagePreviewStore.instance.off(ROOM_PREVIEW_CHANGED, this.onRoomPreviewChanged);
     }
 
+    public shouldComponentUpdate(nextProps: Readonly<IProps>, nextState: Readonly<IState>): boolean {
```

Comment:
this frankly looks identical to the built-in behaviour of PureComponents

> React.PureComponent implements it with a shallow prop and state comparison.

Reply:
See https://reactjs.org/docs/react-api.html#reactpurecomponent

Reply:
indeed, and it looks like their implementation is lot more unstable than ours but there's no point in duplicating the effort.